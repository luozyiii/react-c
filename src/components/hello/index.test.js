import React from 'react';
import Hello from './index';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { expect } from '@jest/globals';

Enzyme.configure({ adapter: new Adapter() });

it('renders without crashing', () => {
  const wrapper = shallow(<Hello />);
  expect(wrapper.find('[data-test="hello"]').length).toBe(1);

  // toExist 是jest-enzyme 的匹配器
  expect(wrapper.find('[data-test="hello"]')).toExist();

  // toHaveText
  expect(wrapper.find('[data-test="hello"]')).toHaveText('hello, this is React Component');

  // 快照
  expect(wrapper).toMatchSnapshot();
});
