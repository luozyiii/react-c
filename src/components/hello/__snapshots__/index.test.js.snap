// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="title"
  data-test="hello"
>
  hello, this is React Component
</div>
`;
